sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -1628,6 +1631,94 @@ def write_pipeline_spec_to_file(pipeline_spec: pipeline_spec_pb2.PipelineSpec,
+        'BOOLEAN': 'bool',
+        'LIST': 'list',
+        'STRUCT': 'dict'
+    }
+
+    def add_inputs():
+        if 'inputDefinitions' in pipeline_spec['root']:
+            inputs = pipeline_spec['root']['inputDefinitions']
+            string = '# Inputs: \n'
+
+            if 'parameters' in inputs:
+                for parameter in inputs['parameters']:
+                    string += '#    ' + parameter + ': ' + map_parameter_types[
+                        inputs['parameters'][parameter]['parameterType']]
+                    if 'defaultValue' in inputs['parameters'][parameter]:
+                        string += ' [Default: ' + str(
+                            inputs['parameters'][parameter]
+                            ['defaultValue']) + ']'
+
+                    string += '\n'
+
+            if 'artifacts' in inputs:
+                for artifact in inputs['artifacts']:
+                    if 'schemaTitle' in inputs['artifacts'][artifact][
+                            'artifactType']:
+                        string += '#    ' + artifact + ': ' + inputs[
+                            'artifacts'][artifact]['artifactType'][
+                                'schemaTitle'] + '\n'
+                    #TODO: Add exception tp raise if schematitle doesnt exist
+
+            return string
+        else:
+            return ''
+
+    def add_outputs():
+        if 'outputDefinitions' in pipeline_spec['root']:
+            outputs = pipeline_spec['root']['outputDefinitions']
+            string = '# Outputs: \n'
+
+            if 'parameters' in outputs:
+                for parameter in outputs['parameters']:
+                    string += '#    ' + parameter + ': ' + map_parameter_types[
+                        outputs['parameters'][parameter]['parameterType']]
+                    if 'defaultValue' in outputs['parameters'][parameter]:
+                        string += ' [Default: ' + str(
+                            outputs['parameters'][parameter]
+                            ['defaultValue']) + ']'
+
+                    string += '\n'
+
+            if 'artifacts' in outputs:
+                for artifact in outputs['artifacts']:
+                    if 'schemaTitle' in outputs['artifacts'][artifact][
+                            'artifactType']:
+                        string += '#    ' + artifact + ': ' + outputs[
+                            'artifacts'][artifact]['artifactType'][
+                                'schemaTitle'] + '\n'
+
+                    #TODO: Add exception tp raise if schematitle doesnt exist
+
+            return string
+        else:
+            return ''
+
+    if 'root' not in pipeline_spec:
+        return ''
+
+    comment = '# PIPELINE DEFINITION\n'
+    comment += '# Name: ' + pipeline_spec['pipelineInfo']['name'] + '\n'
+    if pipeline_description:
+        comment += '# Description: ' + pipeline_description + '\n'
+    comment += add_inputs()
+    comment += add_outputs()
```

Comment:
nit: Consider obtaining all the different sections in a list, then using a string joining method. I think this would slightly improve readability and reduce the likelihood that a `\n` is omitted on one of the strings somewhere.

`'\n'.join(sections)`

The same pattern might be helpful in `add_inputs` and `add_outputs`